### ISSUE TITLE
Stop monkey patching bn.js

### ISSUE THREAD

**Issue by paddyquinn** [bitpay/bitcore#2190]
Noticed this in [`dashcore-lib`](https://github.com/dashevo/dashcore-lib), which is a fork of this lib. The [`bn.js`](https://github.com/bitpay/bitcore/blob/master/packages/bitcore-lib/lib/crypto/bn.js) file within this lib pulls in the [`bn.js`](https://www.npmjs.com/package/bn.js) npm module and then dynamically overrides its methods. This makes for a very frustrating developer experience. More details can be found in the [issue](https://github.com/dashevo/dashcore-lib/issues/125) I opened in `dashcore-lib`.

**Comment by micahriggan** [comment]
Agreed, there's a few things in bitcore-lib where we're polluting other modules. An MR to fix that would be greatly appreciated.

**Comment by HoOngEe** [comment]
Agreed, I also went through an absurd experience. When after importing I tried `new BN("00cd667f8287be106305f9fa3e2a1706579a179b084dc8247c0e31ebc900df9d", 16).toBuffer("be", 32).length`, it returns 31!
Because this library modifies the method parametrization, I have to use `new BN("00cd667f8287be106305f9fa3e2a1706579a179b084dc8247c0e31ebc900df9d", 16).toBuffer({endian: "big", size: 32).length` instead to get intended result.
